src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -391,18 +391,37 @@ public override void VisitMethod(IMethodSymbol symbol)
                     var name = format.CompilerInternalOptions.IncludesOption(SymbolDisplayCompilerInternalOptions.UseMetadataMethodNames) || symbol.ContainingType == null || symbol.ContainingType.IsAnonymousType
                         ? symbol.Name
                         : symbol.ContainingType.Name;
-                    builder.Add(CreatePart(SymbolDisplayPartKind.MethodName, symbol, name));
+
+                    // In the case that symbol.containingType is null (which should never be the case here) we will fallback to the MethodName symbol part
+                    var displayPartKind = SymbolDisplayPartKind.MethodName;
+                    if (symbol.ContainingType != null)
+                    {
+                        displayPartKind = symbol.ContainingType.TypeKind == TypeKind.Class
+                            ? SymbolDisplayPartKind.ClassName
+                            : SymbolDisplayPartKind.StructName;
+                    }
+
+                    builder.Add(CreatePart(displayPartKind, symbol, name));
```

Comment:
>builder.Add(CreatePart(displayPartKind, symbol, name)); [](start = 20, length = 55)

Please add tests to `Microsoft.CodeAnalysis.CSharp.UnitTests.SymbolDisplayTests`.